### ISSUE TITLE
Send before ready

### ISSUE THREAD

**Issue by victorherraiz** [SOHU-Co/kafka-node#989]
I notice that calling `Producer.send` before `ready` event deliver the message successfully. 

* is this the expected behaviour?
* is this recommended?

Thank you

**Comment by hyperlink** [comment]
It's possible this could work with KafkaClient but I would still wait for the ready first.

**Comment by victorherraiz** [comment]
Yes, I am using KafkaClient in the development environment and, I hope, I will use ZooKeeper in production. The use case is an audit library for more than 50 microservices. 

I have some problems with the design and understanding the flow I should use. 

As an example of the library usage:

```js
const audit = require('@something/audit')
 // several lines omitted

audit.config().then((middleware) => {
  // Configure express application
  // several lines omitted
   app.use(middleware)
}).catch((error) => {
  // stop, log and exit
}
```

As you see, I want to reject the promise is the connection fails or kafka is not available.

Should I listen once to error and once to ready?  or is better a timeout?
